articles/events_executor.md

```diff
@@ -129,11 +132,63 @@ This class will forward the items needed for pushing events to the notify guard
 Whenever an entity is associated to an executor it may already have multiple items of work ready to be processed.
 The `EventsExecutor` should tell the entity whether it has to immediately push or to discard those pending events.
 
+### Types of Events queue
+
+There are multiple uses cases with different requirements (performance, determinism in events ordering, bounded memory). A single `EventsQueue` can't comply with all requirements simultaneously, so there's need for different types of queue.
```

Comment:
We should focus on describing the design of the base class.

Reply:
let's have a small paragraph here listing the possible type of queues that could be implemented

Reply:
Done in 0d70bf2